PWG/EMCAL/EMCALbase/AliAnalysisTaskEmcalEmbeddingHelperData.cxx

```diff
@@ -0,0 +1,97 @@
+#include <vector>
+
+#include "AliAnalysisTaskEmcalEmbeddingHelper.h"
+#include "AliAnalysisTaskEmcalEmbeddingHelperData.h"
+#include "AliAODEvent.h"
+#include "AliESDEvent.h"
+#include "AliVAODHeader.h"
+#include "AliEmcalList.h"
+
+/// \cond CLASSIMP
+ClassImp(AliAnalysisTaskEmcalEmbeddingHelperData);
+/// \endcond
+
+AliAnalysisTaskEmcalEmbeddingHelperData::AliAnalysisTaskEmcalEmbeddingHelperData():
+  AliAnalysisTaskEmcalEmbeddingHelper(),
+  fHistManager(0),
+  fPtSelectionMin(),
+  fPtSelectionMax()
+{
+}
+
+
+AliAnalysisTaskEmcalEmbeddingHelperData::AliAnalysisTaskEmcalEmbeddingHelperData(const char *name):
+  AliAnalysisTaskEmcalEmbeddingHelper(name),
+  fHistManager(name),
+  fPtSelectionMin(),
+  fPtSelectionMax()
+{
+}
+
+AliAnalysisTaskEmcalEmbeddingHelperData::~AliAnalysisTaskEmcalEmbeddingHelperData()
+{
+}
+
+void AliAnalysisTaskEmcalEmbeddingHelperData::UserCreateOutputObjects() 
+{
+
+  // run function from embedding helper base class
+  AliAnalysisTaskEmcalEmbeddingHelper::UserCreateOutputObjects();
+
+  TString histName = "fHistPtSelection";
+  TString histTitle = "pT selection;Result;p_{T}^{lead}";
+  auto histPtSelection= fHistManager.CreateTH2(histName, histTitle, 2, 0, 2, 200, 0, 100);
+  histPtSelection->GetXaxis()->SetBinLabel(1,"Accepted");
+  histPtSelection->GetXaxis()->SetBinLabel(2,"Rejected");
+
+  // Add all histograms to output list
+  TIter next(fHistManager.GetListOfHistograms());
+  TObject* obj = 0;
+  while ((obj = next())) {
+    fOutput->Add(obj);
+  }
+
+  PostData(1, fOutput);
+
+
+}
+
+/**
+ * Override inherited function and add check on track with max pT.
+ * Rejects event if track within set pT range is not found
+ */
+
+Bool_t AliAnalysisTaskEmcalEmbeddingHelperData::CheckIsEmbeddedEventSelected()
+{
```

Comment:
You may not need this at the moment, but you might also consider adding an AliEventCuts member so that we can perform event selection on the pp data. I don't know if this is super critical, but something to keep in mind. You can see this is implemented for the internal event in the embedding helper base class